### PR TITLE
Submit test coverage info to code climate and codecov

### PR DESCRIPTION
**Pull request by jim**
## Description

This PR adds a new job to our CircleCI config that runs the Go unit tests and submits the coverage information to [CodeClimate](https://codeclimate.com) and [codecov](https://codecov.io).

I'd like to merge this in a let it run for a little while to see if any of the tools on these services prove to be useful.

## Reviewer Notes

The `cc-test-reporter` program is a little picky about where it expects things to be, and in order to get everything working some commands need to be run from a directory above the main repo. I think this is harmless as it only affects this job. `codecov` was a lot easier to configure, FWIW.

I haven't gotten into client-side test coverage, but I will look into that next.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -207,14 +207,14 @@ server_test_all: server_deps server_generate db_dev_reset db_dev_migrate
 	# Like server_test but runs extended tests that may hit external services.
 	go test -p 1 -count 1 $$(go list ./... | grep -v \\/pkg\\/gen\\/ | grep -v \\/cmd\\/)
 
-server_test_coverage: server_deps server_generate db_dev_reset db_dev_migrate
+server_test_coverage: server_deps server_generate db_test_reset db_test_migrate
```
Comment (jim) [inline Makefile]: This change was needed to get `make server_test_coverage` to run.

File: Makefile [hunk 2]
```diff
@@ -207,14 +207,14 @@ server_test_all: server_deps server_generate db_dev_reset db_dev_migrate
...
 	# Don't run tests in /cmd or /pkg/gen
 	# Use -test.parallel 1 to test packages serially and avoid database collisions
 	# Disable test caching with `-count 1` - caching was masking local test failures
 	# Add coverage tracker via go cover
 	# Then open coverage tracker in HTML
-	go test -coverprofile=coverage.out -p 1 -count 1 $$(go list ./... | grep -v \\/pkg\\/gen\\/ | grep -v \\/cmd\\/)
-	go tool cover -html=coverage.out
+	go test -coverprofile=coverage.out -covermode=count -p 1 -count 1 -short $$(go list ./... | grep -v \\/pkg\\/gen\\/ | grep -v \\/cmd\\/)
```
Comment (jim) [inline Makefile]: `-covermode=count` counts how many times each line is run, not only if it was run. [Read more](https://blog.golang.org/cover#TOC_6.).

File: Makefile [hunk 3]
```diff
@@ -207,14 +207,14 @@ server_test_all: server_deps server_generate db_dev_reset db_dev_migrate
...
+	# go tool cover -html=coverage.out
```
Comment (jim) [inline Makefile]: This is useful for developers, so I can move it to another make target.

File: Makefile (hunk 2 above)
Comment (chrisgilmerproj) [reply Makefile]: Do we still want to remove the comment block? The first couple lines seem relevant still.

File: Makefile (hunk 3 above)
Comment (chrisgilmerproj) [reply Makefile]: 👍

File: Makefile (hunk 2 above)
Comment (jim) [reply Makefile]: I put it back!

Comment (chrisgilmerproj) [reply Makefile]: Thanks! 💯

**Pull request merged**